### PR TITLE
CRM: Update the workflow schema to use pointer steps

### PR DESCRIPTION
**Pull request by cleacos**
Fixes https://github.com/Automattic/zero-bs-crm/issues/3312

## Proposed changes:

This PR updates the Workflow schema and the steps inside of it. Now, steps are linked through pointers, with the step id.

Example of Workflow:

```
array(
    'id'           => 'testing',
    'name'         => 'New Contact',
    'active'       => true,
    'version'      => 1,
    'description'  => 'This automation will change the status of a contact to "Customer" when they are created.',
    'category'     => 'Contact',
    'triggers'     => array(
        'jpcrm/invoice_created',
    ),
    'initial_step' => 'step_1',
    'steps'        => array(
        'step_1' => array(
            'slug'       => 'jpcrm/update_contact_status',
            'attributes' => array(
                'new_status' => 'Customer',
            ),
            'next_step'  => 'step_2',
        ),
        'step_2' => array(
            'slug'       => 'jpcrm/send_email',
            'attributes' => array(
                'to'      => 'admin@example.com',
                'subject' => 'New Customer',
                'body'    => 'A new customer has been created.',
            ),
            'next_step'  => null,
        ),
    ),
);
```
Now, instead of having the `initial_step` with all the step info linked (folder scheme), we have the new `steps` array with an index (step ID), and the `initial_step` has the step id (index). Also, `next_step` `next_step_true` `next_step_false` is the ID of the next step.

**Note**: We should work in our model with the Step instance rather than retrieving IDs. This will be handled in a following PR.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pbhBOz-3Wf-p2

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Check the code.
- Check that the tests continue passing.

### CODE REVIEW

File: projects/plugins/crm/src/automation/class-automation-engine.php [hunk 1]
```diff
@@ -293,8 +293,8 @@ public function register_step( string $class_name ) {
 			);
 		}
 
-		$step_name                     = $class_name::get_slug();
-		$this->steps_map[ $step_name ] = $class_name;
+		$step_slug                     = $class_name::get_slug();
+		$this->steps_map[ $step_slug ] = $class_name;
```
Comment (gogdzl) [inline projects/plugins/crm/src/automation/class-automation-engine.php:297]: Nice change!

**Review by gogdzl — APPROVED** [review]
The only part I find confusing is the fact that we allow strings and integers as valid step_ids.

I just would like some clarification on that matter. Otherwise, it works!

**Comment by cleacos** [comment]
> The only part I find confusing is the fact that we allow strings and integers as valid step_ids.

I left it open because mainly, and for now, workflow steps are not going to be stored alone in DB, so that is just an index we have to set and it could be `1`, `2`,... or use another format like `s1`, `s2`,... or `step_1`, `step_2`. Better to read it in an array. So because steps are not as storable as a DB entity, we don't have the unique ID concept here is just an index inside of the workflow.

**Comment by cleacos** [comment]
Merging it to trunk...

**Pull request merged**